File: public/customize/server/utils/pdf_generator.py
from reportlab.lib.pagesizes import letter
from reportlab.pdfgen import canvas
from reportlab.lib import colors
import os
import datetime

def generate_invoice(order):
    invoices_dir = os.path.join(os.path.dirname(__file__), '../invoices')
    if not os.path.exists(invoices_dir):
        os.makedirs(invoices_dir)
        
    file_name = f"invoice_{order['_id']}.pdf"
    file_path = os.path.join(invoices_dir, file_name)
    
    c = canvas.Canvas(file_path, pagesize=letter)
    width, height = letter

    # Header
    c.setFont("Helvetica-Bold", 25)
    c.drawCentredString(width/2, height - 50, "KNIGHT WOLF")
    
    c.setFont("Helvetica", 10)
    c.drawCentredString(width/2, height - 65, "STREETWEAR FOR THE ALPHAS")
    
    c.setLineWidth(2)
    c.line(50, height - 80, 550, height - 80)

    # Order Info
    c.setFont("Helvetica", 12)
    c.setFillColor(colors.HexColor("#444444"))
    c.drawString(50, height - 110, f"Order ID: {order['_id']}")
    
    date_str = datetime.datetime.fromisoformat(order['createdAt']).strftime('%Y-%m-%d')
    c.drawString(50, height - 125, f"Date: {date_str}")
    c.drawString(50, height - 140, f"Customer ID: {order['userId']}")
    
    c.setFillColor(colors.green)
    c.drawString(50, height - 155, "Status: PAID")
    
    # Table Header
    c.setFillColor(colors.black)
    c.setFont("Helvetica-Bold", 12)
    table_top = height - 190
    c.drawString(50, table_top, "Item")
    c.drawString(250, table_top, "Size")
    c.drawString(350, table_top, "Qty")
    c.drawString(450, table_top, "Price")
    
    c.setLineWidth(1)
    c.line(50, table_top - 5, 550, table_top - 5)
    
    # Items
    c.setFont("Helvetica", 11)
    c.setFillColor(colors.HexColor("#333333"))
    current_y = table_top - 25
    
    for item in order['items']:
        c.drawString(50, current_y, item['name'])
        c.drawString(250, current_y, item['size'])
        c.drawString(350, current_y, str(item['quantity']))
        c.drawString(450, current_y, f"Rs.{item['price']}")
        current_y -= 20
        
    # Total
    c.setLineWidth(2)
    c.line(50, current_y - 10, 550, current_y - 10)
    
    c.setFont("Helvetica-Bold", 15)
    c.drawRightString(550, current_y - 30, f"TOTAL AMOUNT: Rs.{order['totalAmount']}")
    
    # Footer
    c.setFont("Helvetica-Oblique", 12)
    c.drawCentredString(width/2, 100, "Thank you for joining the Wolf Pack.")
    
    c.save()
    return file_path

File: public/customize/server/utils/pdfGenerator.js
const PDFDocument = require('pdfkit');
const fs = require('fs');
const path = require('path');

const generateInvoice = (order) => {
    return new Promise((resolve, reject) => {
        const doc = new PDFDocument({ margin: 50 });
        const fileName = `invoice_${order._id}.pdf`;
        const filePath = path.join(__dirname, '../invoices', fileName);

        // Ensure invoices directory exists
        const dir = path.join(__dirname, '../invoices');
        if (!fs.existsSync(dir)) {
            fs.mkdirSync(dir);
        }

        const stream = fs.createWriteStream(filePath);
        doc.pipe(stream);

        // Header
        doc.fillColor('#000000').fontSize(25).font('Helvetica-Bold').text('KNIGHT WOLF', { align: 'center' });
        doc.fontSize(10).font('Helvetica').text('STREETWEAR FOR THE ALPHAS', { align: 'center' });
        doc.moveDown();
        doc.rect(50, doc.y, 500, 2).fill('#000000');
        doc.moveDown();

        // Order Info
        doc.fillColor('#444444').fontSize(12).text(`Order ID: ${order._id}`);
        doc.text(`Date: ${new Date(order.createdAt).toLocaleDateString()}`);
        doc.text(`Customer ID: ${order.userId}`);
        doc.text(`Status: PAID`, { color: '#008000' });
        doc.moveDown();

        // Table Header
        const tableTop = doc.y;
        doc.font('Helvetica-Bold').fillColor('#000000');
        doc.text('Item', 50, tableTop);
        doc.text('Size', 250, tableTop);
        doc.text('Qty', 350, tableTop);
        doc.text('Price', 450, tableTop);
        doc.moveDown();
        doc.rect(50, doc.y, 500, 1).fill('#cccccc');
        doc.moveDown();

        // Items
        let currentY = doc.y;
        order.items.forEach(item => {
            doc.font('Helvetica').fillColor('#333333');
            doc.text(item.name, 50, currentY);
            doc.text(item.size, 250, currentY);
            doc.text(item.quantity.toString(), 350, currentY);
            doc.text(`₹${item.price}`, 450, currentY);
            currentY += 20;
        });

        // Footer
        doc.rect(50, currentY + 20, 500, 2).fill('#000000');
        doc.fontSize(15).font('Helvetica-Bold').text(`TOTAL AMOUNT: ₹${order.totalAmount}`, 50, currentY + 40, { align: 'right' });

        doc.moveDown(5);
        doc.fontSize(12).font('Helvetica-Oblique').text('Thank you for joining the Wolf Pack.', { align: 'center' });

        doc.end();

        stream.on('finish', () => resolve(filePath));
        stream.on('error', reject);
    });
};

module.exports = { generateInvoice };
